test/unit/math/rev/mat/fun/cholesky_decompose_test.cpp

```diff
@@ -432,3 +437,21 @@ TEST(AgradRevMatrix, check_varis_on_stack_large) {
   X = stan::math::multiply(X, stan::math::transpose(X));
   test::check_varis_on_stack(stan::math::cholesky_decompose(X));
 }
+
+#ifdef STAN_OPENCL
+TEST(AgradRevMatrix, mat_cholesky_1st_deriv_large_gradients_opencl) {
+  stan::math::opencl_context.tuning_opts().cholesky_size_worth_transfer = 25;
+  test_gradient(51, 1e-08);
+  test_gp_grad(1300, 1e-08);
+  test_gp_grad(2000, 1e-08);
+  test_chol_mult(50, 1e-08);
+  test_simple_vec_mult(45, 1e-08);
+}
+
+TEST(AgradRevMatrix, check_varis_on_stack_large_opencl) {
+  stan::math::matrix_v m1 = stan::math::matrix_v::Random(1250, 1250);
+  stan::math::matrix_v m1_pos_def
+      = m1 * m1.transpose() + 500 * stan::math::matrix_v::Identity(1250, 1250);
+  test::check_varis_on_stack(stan::math::cholesky_decompose(m1_pos_def));
+}
+#endif
```

Comment:
Sean we don't have tests here for exceptions, my thought process here was that since those checks go off for gpu and non-gpu code they would be redundant to include for larger matrices